public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -597,7 +599,20 @@ function errorAndDurationQuery(
   let serviceGraphViewMetrics = [];
   let errorRateBySpanName = '';
   let durationsBySpanName: string[] = [];
-  const spanNames = getEscapedSpanNames(rateResponse.data[0][0]?.fields[1]?.values ?? []);
+
+  let labels = [];
+  if (request.app === CoreApp.Explore) {
+    if (rateResponse.data[0][0]?.fields[1]?.values) {
```

Comment:
Why is the response different for explore vs dashboard?

Reply:
It seems that in dashboards there's multiple data frames returned, each with span_name label. In explore some sort of transform is applied and all are aggregated into a single data frame, with span_name field and span names as values

Reply:
Interesting, do you know why is that? I know we have different formats for time series, with either the wide format or dataFrame per series but for trace, to have a dataFrame per span seems a bit inefficient (especially if we have traces with thousands of spans)

Reply:
oh right, this is actually a prometheus query that returns the nodegraph data so I assume this is exactly the case that prometheus returns a different shape. In that case span_name is basically a node name and so we probably don't really have that many nodes? I mean maybe the naming is weird, would expect the label to be `service_name` probably not `span_name` as there can be more spans per single service but we want to show just the service in the service map

Reply:
That's basically it. I don't think we can have that many nodes really and have not seen a case where we do. Agreed the naming is a bit weird but for now I think it's ok. Not exactly sure where we would change the `span_name` or if it's in the Tempo metrics generator/Prometheus code, honestly it's probably not necessary right now.